src/pytorch_lightning/plugins/io/wrapper.py

```diff
@@ -0,0 +1,68 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Dict, Optional
+
+from pytorch_lightning.plugins.io.checkpoint_plugin import CheckpointIO
+
+
+class _WrappingCheckpointIO(CheckpointIO):
+    """AsyncCheckpointIO enables saving the checkpoints asynchronously.
+
+    .. warning::
+
+        This is currently an experimental plugin/feature and API changes are to be expected.
+
+    Args:
+        checkpoint_io: A checkpoint IO plugin that is used as the basis for async checkpointing.
+        interval: Sleep time between each queue check.
+    """
+
+    def __init__(self, checkpoint_io: Optional["CheckpointIO"] = None) -> None:
+        super().__init__()
+
+        self._checkpoint_io = checkpoint_io
```

Comment:
IMO this should not be an Optional attribute, meaning that if `None` was passed, we default to `TorchCheckpointIO`

Reply:
then it would be hard for us to tell which checkpoint io to use for eg. let's say when the user is using HPU, we can't tell whether it is passed by the user or is the default one unless we add some assertions or boolean flags...

Just wanted to keep it a little agnostic and let strategies decide the default one on their own.

Reply:
I have to agree with @rohitgr7 that None is a good default value to differentiate for the cases where the `checkpoint_io` hasn't been set explicitly and has to be set by strategies.